core/src/main/scala/org/apache/spark/scheduler/ExecutorLossReason.scala

```diff
@@ -37,6 +37,8 @@ private[spark] object ExecutorExited {
   }
 }
 
+private [spark] object LossReasonPending extends ExecutorLossReason("Pending loss reason.")
```

Comment:
It took me a little bit to realize what this means. I think something like `WaitingForLossReason` or `WaitingForRealLossReason` is clearer. Then we should add a java doc to explain why we need to wait for loss reason.